common/src/main/java/no/difi/meldingsutveksling/pipes/Pipe.java

```diff
@@ -0,0 +1,99 @@
+package no.difi.meldingsutveksling.pipes;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+@Slf4j
+public class Pipe {
+
+    private final Executor executor;
+    private final PipedOutputStream inlet;
+    private final PipedInputStream outlet;
+
+    private Pipe(Executor executor) {
+        this.executor = executor;
+        this.inlet = new PipedOutputStream();
+        this.outlet = new PipedInputStream();
+        connectInletAndOutlet();
+    }
+
+    private void connectInletAndOutlet() {
+        try {
+            inlet.connect(outlet);
+        } catch (IOException e) {
+            throw new PipeRuntimeException("Connect failed!", e);
+        }
+    }
+
+    public PipedInputStream outlet() {
+        return outlet;
+    }
+
+    private void close() {
+        try {
+            inlet.flush();
+            inlet.close();
+        } catch (IOException e) {
+            throw new PipeRuntimeException("Could not close of", e);
+        }
+    }
+
+    private void handleCompleteAsync(Void dummy, Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "dummy". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1172)